src/test/java/domen/infrastructure/database/TaskRepositorySQLTest.java

```diff
@@ -46,24 +44,78 @@ void shouldSaveTaskCorrectly() throws Exception {
         when(mockConnection.prepareStatement(any(String.class))).thenReturn(mockPreparedStatement);
         when(mockPreparedStatement.executeUpdate()).thenReturn(1);
         TaskRepositorySQL database = new TaskRepositorySQL(mockConnection);
-
         //when
         database.save(task);
-
         // then
         ArgumentCaptor<String> sqlCaptor = ArgumentCaptor.forClass(String.class);
-
         verify(mockConnection).prepareStatement(sqlCaptor.capture());
         verify(mockPreparedStatement).executeUpdate();
-
         assertEquals("INSERT INTO task (id, title, description, status, start_date_time, finish_date_time)" +
                 " VALUES (?::uuid,?,?,?::task_status,?::timestamp,?::timestamp)", sqlCaptor.getValue());
-
         verify(mockPreparedStatement).setObject(eq(1), any(UUID.class));
         verify(mockPreparedStatement).setString(eq(2), eq(task.title()));
         verify(mockPreparedStatement).setString(eq(3), eq(task.description()));
         verify(mockPreparedStatement).setString(eq(4), eq(task.status().toString()));
         verify(mockPreparedStatement).setTimestamp(eq(5), eq(Timestamp.valueOf(task.startDateTime())));
         verify(mockPreparedStatement).setTimestamp(eq(6), eq(Timestamp.valueOf(task.finishDateTime())));
     }
+
+    @Test
+    void shouldReturnTaskWhenFoundById() throws SQLException {
+        // given
+        String taskId = UUID.randomUUID().toString();
+        Connection connection = mock(Connection.class);
+        PreparedStatement ps = mock(PreparedStatement.class);
+        ResultSet rs = mock(ResultSet.class);
+        when(connection.prepareStatement(any(String.class))).thenReturn(ps);
+        when(ps.executeQuery()).thenReturn(rs);
+        when(rs.next()).thenReturn(true);
+        when(rs.getObject("id", UUID.class)).thenReturn(UUID.fromString(taskId));
+        when(rs.getString("title")).thenReturn("Test Title");
+        when(rs.getString("description")).thenReturn("Test Description");
+        when(rs.getString("status")).thenReturn("NEW");
+        when(rs.getTimestamp("start_date_time")).thenReturn(Timestamp.valueOf(LocalDateTime.of(2024, 4, 26, 12, 0)));
+        when(rs.getTimestamp("finish_date_time")).thenReturn(Timestamp.valueOf(LocalDateTime.of(2024, 4, 26, 14, 0)));
+        TaskRepositorySQL database = new TaskRepositorySQL(connection);
+        //when
+        Optional<Task> result = database.findById(taskId);
+        //then
+        assertTrue(result.isPresent());
+        Task task = result.get();
+        assertEquals("Test Title", task.title());
```

Comment:
- [ ] 2 razy magic string Ci sie pojawia (np. tutaj i w 74 linii), chwile mi zajelo zeby skumac skad sie bierze akurat ta wartosc. Proponuje bys sobie w given stworzyl Task i wykorzystywal wartosci jego pol w mockito.when oraz then

- [ ] nie porównujesz tutaj wartości sql'ki pobrany z ArgumentCaptor, czy to zamierzony efekt?